packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/Carousel/CarouselItem.tsx

```diff
@@ -0,0 +1,25 @@
+import { useContextSelector } from '@fluentui/react-context-selector';
+import { CarouselContext } from './useCarouselCollection';
+import * as React from 'react';
+import { CarouselItemProps } from './Carousel.types';
+
+// TODO: We are going to move this to it's own <CarouselItem/> component, for now we use a simple function to wrap.
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export function CarouselItem_unstable(props: CarouselItemProps) {
+  const { value } = props;
+
+  const carouselContext = useContextSelector(CarouselContext, c => c);
+
+  if (!carouselContext) {
+    return props.children;
+  }
+
+  const visible = carouselContext.value === value;
```

Comment:
```suggestion
  const visible = useContextSelector(CarouselContext, c => carouselContext.value === value);
```

You can't pull `carouselContext` as it's reference will be unstable, we will should pull only its parts.